openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiSchemaGenericsSpec.groovy

```diff
@@ -265,4 +263,158 @@ class MyBean {}
+    Second(BigDecimal.ONE, "s"),
+    Minute(BigDecimal.valueOf(60), Second, "m"),
+    Hour(BigDecimal.valueOf(60), Minute, "h"),
+    Day(BigDecimal.valueOf(24), Hour, "d"),
+    Week(BigDecimal.valueOf(7), Day, "w");
+
+    private final BigDecimal ratio;
+    private final String suffix;
+
+    TimeUnit(BigDecimal ratio, String suffix) {
+        this.ratio = ratio;
+        this.suffix = suffix;
+    }
+
+    TimeUnit(BigDecimal factor, TimeUnit base, String suffix) {
+        this.ratio = factor.multiply(base.ratio);
+        this.suffix = suffix;
+    }
+
+    @Override public BigDecimal ratio() {
+        return ratio;
+    }
+
+    @Override public String suffix() {
+        return suffix;
+    }
+
+}
+
+interface Unit {
+
+    MathContext MATH_CONTEXT = new MathContext(16, HALF_UP);
+
+    String name();
+
+    BigDecimal ratio();
+
+    String suffix();
+}
+
+class Time extends Quantity<Time, TimeUnit> {
+
+    private Time(BigDecimal amount, TimeUnit unit) {
+        super(amount, unit);
+    }
+
+    public static Time of(BigDecimal amount, TimeUnit unit) {
+        return new Time(amount, unit);
+    }
+
+    @Override
+    public BigDecimal getAmount() {
+        return super.getAmount();
+    }
+
+    @Override
+    public TimeUnit getUnit() {
+        return super.getUnit();
+    }
+
+    @Override
+    public void setAmount(BigDecimal amount) {
+        super.setAmount(amount);
+    }
+
+    @Override
+    public void setUnit(TimeUnit unit) {
+        super.setUnit(unit);
+    }
+
+}
+
+class Quantity<Q extends Quantity, U extends Unit> implements Serializable {
+
+    private static final long serialVersionUID = -9000608810227353935L;
+
+    private final BigDecimal amount;
+    private final U unit;
+
+    Quantity(BigDecimal amount, U unit) {
+        this.amount = amount;
+        this.unit = unit;
+    }
+
+    public BigDecimal getAmount() {
+        return amount;
+    }
+
+    public U getUnit() {
+        return unit;
+    }
+
+    public void setUnit(U unit) {
+        throw new UnsupportedOperationException("Quantities can't change");
+    }
+
+    public void setAmount(BigDecimal amount) {
+        throw new UnsupportedOperationException("Quantities can't change");
+    }
+}
+
+@javax.inject.Singleton
+class MyBean {}
+''')
+
+        OpenAPI openAPI = AbstractOpenApiVisitor.testReference
+        Operation operation = openAPI.paths?.get("/time")?.get
+
+        expect:
+        operation
+        operation.responses.size() == 1
+        operation.responses.default.content['application/json'].schema.$ref == '#/components/schemas/Time'
+        openAPI.components.schemas['Quantity<Time,TimeUnit>'].type == 'object'
+        openAPI.components.schemas['Quantity<Time,TimeUnit>'].properties['amount'].type == 'number'
+        openAPI.components.schemas['Quantity<Time,TimeUnit>'].properties['unit'].$ref == '#/components/schemas/TimeUnit'
```

Comment:
I think this may need to change after merging the PR that changes `<` to `_`

Reply:
Done